concourse/templates/default.mako

```diff
@@ -213,6 +213,70 @@ if secret_cfg:
 % endif
 </%def>
 
+<%def name="on_abort(indent, job_variant, job_step)" filter="indent_func(indent),trim">
+<%
+if job_variant.has_main_repository():
```

Comment:
factor-out source_repo-handling (I assume this is copy-pasta from regular build step?)?